### PR TITLE
chart.js: add visibility members

### PR DESCRIPTION
**Pull request by Ilmarinen100**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/chartjs/Chart.js/blob/master/src/core/core.controller.js#L848
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by lekoaf** [comment]
Why stop there? We're also missing `get...` and `toggle...`.

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -52,6 +53,8 @@ declare class Chart {
     getElementsAtXAxis: (e: any) => Array<{}>;
     getDatasetAtEvent: (e: any) => Array<{}>;
     getDatasetMeta: (index: number) => Meta;
+    isDatasetVisible: (datasetIndex: number) => boolean;
+    setDatasetVisibility: (datasetIndex: number, visible: boolean) => void;
```
Comment (lekoaf) [inline types/chart.js/index.d.ts:57]: Why stop there? We're also missing `get..`. and `toggle...`.

Comment (Ilmarinen100) [reply types/chart.js/index.d.ts:58]: @lekoaf: quite a lot more are missing and not just for this class :/
I wanted my first contribution to be small (and manageable as a side task - took me long enough to figure out how to properly do the testing - none of the existing tests for this package had used any '// $Expect...' so far)
I added the other visibility related methods now but they are somewhat strange and I have not used them in practice.

Comment (lekoaf) [reply types/chart.js/index.d.ts:58]: What does `// $Expect` do?

Comment (adripanico) [reply types/chart.js/index.d.ts:58]: > What does `// $Expect` do?

I was asking myself exactly the same

Comment (Ilmarinen100) [reply types/chart.js/index.d.ts:58]: nothing it turns out :/ I didn't do a negative check and simply assumed it was like another test framework I used. I am somewhat surprised that the tests do not appear to actually test what is begin given back but only what the typings say - why do the heavy lifting of executing chart.js in a DOM when the types are not verified?

Comment (Ilmarinen100) [reply types/chart.js/index.d.ts:58]: > nothing it turns out :/ I didn't do a negative check and simply assumed it was like another test framework I used. I am somewhat surprised that the tests do not appear to actually test what is begin given back but only what the typings say - why do the heavy lifting of executing chart.js in a DOM when the types are not verified?

... I re-read the documentation that the test is not actually run, my bad

Comment (Ilmarinen100) [reply types/chart.js/index.d.ts:58]: > Why stop there? We're also missing `get..`. and `toggle...`.
I had another look, the members you mentioned are not actually part of chart.js@2.9.x - they are in chart.js 3 that is still in "alpha/pre-release" stage, so I removed them again
* https://github.com/chartjs/Chart.js/blob/2.9/src/core/core.controller.js
* https://github.com/chartjs/Chart.js/blob/release-3.x/src/core/core.controller.js

Comment (lekoaf) [reply types/chart.js/index.d.ts:58]: What? They have alpha/pre-release stuff on the master branch?? Hmm, oh well.

**Review by lekoaf — APPROVED** [review]

**Comment by Ilmarinen100** [comment]
Ready to merge

**Pull request merged**